tsdb/querier.go

```diff
@@ -314,6 +314,18 @@ func PostingsForMatchers(ix IndexReader, ms ...*labels.Matcher) (index.Postings,
 	}
 
 	for _, m := range ms {
+		if m.Type == labels.MatchRegexp && (m.Value == ".*" || m.Value == "^.*$") && len(ms) > 1 {
+			// Ignore this matcher completely. This matches any value, including no value, so it's a no-op.
+			// It's safe to ignore, because there must be some matcher matching non-empty string.
+			// Some tests only use single label=~".*" matcher, and for those, we include the length condition.
```

Comment:
It's not just about tests, it's about correctness. Can we fast path those?

Reply:
You're right. I've modified the code to use all postings if we find this kind of matcher. We only return all postings, if there are no other positive matchers.